src/kudu/util/hdr_histogram.cc

```diff
@@ -155,6 +155,7 @@ void HdrHistogram::Increment(int64_t value) {
 }
 
 void HdrHistogram::IncrementBy(int64_t value, int64_t count) {
+  shared_lock<rw_spinlock> lock(histogram_mutex_);
```

Comment:
Same here, every increment of a metric takes a lock internally. Might be worth to explore a non-locking mechanism for this.

Reply:
But this is shared_lock, so should be ok even if multiple updates are going in parallel.